cli/command/run/cmd.go

```diff
@@ -84,28 +90,60 @@ func Run(cmd *cobra.Command, cli *command.SauceCtlCli, args []string) (int, erro
 		p.Suites = []config.Suite{newDefaultSuite(p)}
 	}
 
-	seq := memseq.Sequencer{}
-
-	r, err := newRunner(p, cli, &seq)
+	r, err := newRunner(p, cli)
 	if err != nil {
 		return 1, err
 	}
 	return r.RunProject()
 }
 
-func newRunner(p config.Project, cli *command.SauceCtlCli, seq fleet.Sequencer) (runner.Testrunner, error) {
+func newRunner(p config.Project, cli *command.SauceCtlCli) (runner.Testrunner, error) {
 	// return test runner for testing
 	if p.Image.Base == "test" {
 		return mocks.NewTestRunner(p, cli)
 	}
 
 	if ci.IsAvailable() {
+		enableCIProviders()
+		// We know we are in a CI environment, but now we try to detect which one
+		cip := ci.Detect()
+		u := os.Getenv("SAUCE_USERNAME")
+		k := os.Getenv("SAUCE_ACCESS_KEY")
+		seq := createCISequencer(p, u, k, cip)
```

Comment:
Given this is only applicable for running `saucectl` in your own infra, it might make sense to move it into the ci runner module.